File: 이종준/week9_programmers/p_신고결과받기.js
function solution(id_list, report, k) {
  var answer = new Array(id_list.length);
  answer.fill(0);
  const reportSet = new Set(report);
  const reportSetToArr = [...reportSet];
  const reportObj = {};
  id_list.forEach((id) => {
    reportObj[id] = [];
  });

  reportSetToArr.forEach((report) => {
    const [userId, reportId] = report.split(" ");
    reportObj[reportId].push(userId);
  });

  for (let key in reportObj) {
    if (reportObj[key].length >= k) {
      reportObj[key].forEach((id) => {
        answer[id_list.indexOf(id)]++;
      });
    }
  }

  return answer;
}

File: 고재현/week9_programmers/p_신고 결과 받기.py
def solution(id_list, report, k):
    r_tmp = {}

    num = {}
    ans = {}
    for name in id_list:
        num[name] = 0
        ans[name] = 0

    for i in range(len(report)):
        reporter = report[i].split()[0]
        reportee = report[i].split()[1]

        if reporter in r_tmp.keys():
            if reportee not in r_tmp[reporter]:
                num[reportee] += 1
                r_tmp[reporter].append(reportee)
        else:
            num[reportee] += 1
            r_tmp[reporter] = [reportee]

    for i in num.keys():
        if num[i] >= k:
            for z in r_tmp.keys():
                if i in r_tmp[z]:
                    ans[z] += 1

    answer = [0] * len(id_list)
    for i in ans.keys():
        for j in range(len(id_list)):
            if i == id_list[j]:
                answer[j] = ans[i]
    return answer